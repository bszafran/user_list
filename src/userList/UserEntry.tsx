import React from "react"
import { User as UserModel } from "./model"
import styled from "styled-components/macro"

interface UserEntryProps {
  user: UserModel
  orderNumber: number
}

const UserEntryContainer = styled.div`
  margin-top: 10px;
`

const Faded = styled.span`
  color: #b8b8b8;
`

const NameContainer = styled.span`
  margin: 0 10px;
`

const UserNameContainer = styled(Faded)`
  font-size: 13px;
`

const UserEntry = (props: UserEntryProps) => {
  const { orderNumber } = props
  const { name, username } = props.user

  return (
    <UserEntryContainer>
      <Faded>{orderNumber}.</Faded>
      <NameContainer>{name}</NameContainer>
      <UserNameContainer>{`@${username}`}</UserNameContainer>
    </UserEntryContainer>
  )
}

export default UserEntry
