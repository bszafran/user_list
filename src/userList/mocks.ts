import { User } from "./model"

const mockedUsers: User[] = [
  {
    id: 1,
    name: "Janusz Kowalski",
    username: "januszk",
  },
  {
    id: 3,
    name: "Micha≈Ç Nowak",
    username: "nowakm",
  },
]

export { mockedUsers }
