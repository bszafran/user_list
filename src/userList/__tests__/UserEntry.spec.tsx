import React from "react"
import UserEntry from "../UserEntry"
import { shallow } from "enzyme"
import { mockedUsers } from "../mocks"

describe("UserEntry component", () => {
  const mockedProps = {
    orderNumber: 5,
    user: mockedUsers[0],
  }

  it("renders correctly with mocked props", () => {
    const userEntryComponent = shallow(<UserEntry {...mockedProps} />)
    expect(userEntryComponent).toMatchSnapshot()
  })
})
