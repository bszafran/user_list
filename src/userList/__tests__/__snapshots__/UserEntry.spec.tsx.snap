// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserEntry component renders correctly with mocked props 1`] = `
<UserEntry__UserEntryContainer>
  <UserEntry__Faded>
    5
    .
  </UserEntry__Faded>
  <UserEntry__NameContainer>
    Janusz Kowalski
  </UserEntry__NameContainer>
  <UserEntry__UserNameContainer>
    @januszk
  </UserEntry__UserNameContainer>
</UserEntry__UserEntryContainer>
`;
